File: swift/lib/Sources/Data/MappingData.swift
public class MappingData {
    public static var shared: MappingData = .init()
    private init() {}

    private var list = [MappingItem]()

    public func add(item: MappingItem) {
        list.append(item)
    }

    public func find(_ name: String) -> MappingItem? {
        return list.first(where: { $0.name == name })
    }
}

File: kotlin/lib/library/src/main/java/com/xplpc/data/MappingData.kt
package com.xplpc.data

import com.xplpc.map.MappingItem

object MappingData {
    private val list = ArrayList<MappingItem>()

    fun add(item: MappingItem) {
        list.add(item)
    }

    fun find(name: String): MappingItem? {
        return list.find { it.name == name }
    }
}
